src/components/topo/ucc_sbgp.c

```diff
@@ -203,18 +205,33 @@ static ucc_status_t sbgp_create_node_leaders(ucc_topo_t *topo, ucc_sbgp_t *sbgp,
         return UCC_ERR_NO_MEMORY;
     }
 
+    nl_array_3 = ucc_malloc(comm_size * nnodes * sizeof(ucc_rank_t),
```

Comment:
seems like too much memory, we cannot have number of sockets more than comm_size